src/main/java/org/assertj/core/api/AbstractBigDecimalAssert.java

```diff
@@ -263,6 +264,25 @@ public SELF isNotEqualByComparingTo(String notExpected) {
     return isNotEqualByComparingTo(new BigDecimal(notExpected));
   }
 
+  /**
+   * Verifies the scale of the BigDecimal passed is equal to the given one.
+   * <p>
+   * Example:
+   * <pre><code class='java'> // assertions will pass
+   * assertThat(new BigDecimal(&quot;8.00&quot;)).isScaleEqualTo(2);
+   * assertThat(new BigDecimal(&quot;8.00&quot;).setScale(4)).isScaleEqualTo(4);
+   *
+   * // assertion will fail
+   * assertThat(new BigDecimal(&quot;8.00&quot;)).isScaleEqualTo(3);
+   * assertThat(new BigDecimal(&quot;8.00&quot;).setScale(4)).isScaleEqualTo(2);</code></pre>
+   * @param expected the expected scale value..
+   * @return {@code this} assertion object.
+   */
+  public SELF isScaleEqualTo(int expected) {
```

Comment:
would `hasScale` be nicer?

Reply:
Yes. Sorry for being pedantic but how about `hasScaleOf`?

Reply:
sounds better :+1: